reactive/kotlinx-coroutines-reactive/src/Publish.kt

```diff
@@ -250,4 +265,6 @@ private class PublisherCoroutine<in T>(
         cancelled = true
         super.cancel(null)
     }
+
+    private fun Throwable.isFatal() = this is VirtualMachineError || this is ThreadDeath || this is LinkageError
```

Comment:
This set is specific to RxJava2 – RxJava1 had different behavior, and other reactive frameworks might as well. A [comment in the RxJava2 source](https://github.com/ReactiveX/RxJava/blob/cc7fce89c46e67c4a124da9e206e4c770b6f8850/src/main/java/io/reactivex/exceptions/Exceptions.java#L67) says [this set was taken from Scala's Futures library](https://github.com/ReactiveX/RxJava/issues/748#issuecomment-32471495).

It's probably a reasonable default, but I'm not sure how this will interact with other reactive libraries that implement `Publisher`.

Reply:
Well, there an extended set in reactor, I will separately take care of it during #1138